modin/core/io/text/json_glob_dispatcher.py

```diff
@@ -0,0 +1,99 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Module houses ``JsonGlobDispatcher`` class that is used for reading `.json` files."""
+
+import glob
+import warnings
+
+from modin.core.io.file_dispatcher import FileDispatcher
+from modin.config import NPartitions
+
+
+class JsonGlobDispatcher(FileDispatcher):
+    """Class handles utils for reading `.pkl` files."""
+
+    @classmethod
+    def _read(cls, filepath_or_buffer, **kwargs):
+        """
+        Read data from `filepath_or_buffer` according to `kwargs` parameters.
+
+        Parameters
+        ----------
+        filepath_or_buffer : str, path object or file-like object
+            `filepath_or_buffer` parameter of `read_json` function.
+        **kwargs : dict
+            Parameters of `read_json` function.
+
+        Returns
+        -------
+        new_query_compiler : BaseQueryCompiler
+            Query compiler with imported data for further processing.
+
+        Notes
+        -----
+        In experimental mode, we can use `*` in the filename.
+
+        The number of partitions is equal to the number of input files.
+        """
+        if not (isinstance(filepath_or_buffer, str) and "*" in filepath_or_buffer):
+            warnings.warn("Defaulting to Modin core implementation")
+            return cls.single_worker_read(
```

Comment:
I think this should fallback to `pd.read_json()`, as we can still parallelize reading json file if it's a multiline json.